test/iron-input.html

```diff
@@ -138,6 +163,27 @@
         assert.equal(input.bindValue, '123');
       });
 
+      test('prevent invalid input works in _onInput and does not use pattern', function() {
+        var input = fixture('prevent-invalid-input-with-pattern');
+        input.value = '123';
+        input._onInput();
+        assert.equal(input.bindValue, '123');
+        input.validate();
```

Comment:
It does both test that `validate` is working with `pattern` and that `pattern` is not interfering with `prevent-invalid-input`. Let's go for `inputs can be validated` :)